### PR TITLE
Move RxJava to a separate folder

### PR DESCRIPTION
**Pull request by Tuubz**
## What happened 🤔
We will soon have a new template for LiveData + Coroutines, to keep our project clean we would need to move our current code in to a separate folder first, then later we can add an additional folder for LiveData + Coroutines

## Insight 👀
In terms of the root project structure and files, I tried to follow according to https://github.com/android/architecture-components-samples

Also, I have adapted the CI workflow, to fix the errors.

What's next? Add the LiveData template and improve the CI workflow as @sleepylee suggested below

## PoW (a.k.a Proof Of Work)💪

The RxJavaTemplate is still buildable:

https://user-images.githubusercontent.com/18277915/103605872-9d951e80-4f47-11eb-93f9-c3f328004c4d.mov

### CODE REVIEW

**Comment by Tuubz** [comment]
CI failed :too_sad:

**Comment by sleepylee** [comment]
for the CI to work, I'd suggest to update the workflow and gradle config at the top level a bit...I'm not having the final answer tho, but ideally, I think the top-level gradle must contain the tasks to execute the test on each sub-directory, then workflow can update the test command accordingly, instead of just calling `./gradlew test`, which understood as running at the current dir level. 🤔 

Another thing is about the CI, if we push a branch to, let's say update coroutine, we wouldn't want CI to run the test on `rx-java` directory right? So we need to figure a way to manage this workflow

**Comment by sleepylee** [comment]
I have 1 idea to check for the condition of the branch name and run the corresponding test on each directory: 
- Take an example, we have a feature to merge to `coroutine-livedata/`
- The branch could be `feature/coroutine-livedata/adding-a-new-feature`
- On the workflow, we can write a script to execute the test by checking the pattern of the branch (can be written in ruby or any language in that current CI env supported).
- If there is 2 slash `/ /` then we run the command with the item[1], for example: `./gradlew coroutine-livedata:test`
- If not, we run the normal command: `./gradlew testAll` (need to be updated at the top-level gradle too, possibly this is good for a release CI run).

You can work on the CI part in a separated PR

File: RxJavaTemplate/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+![Build Status](https://www.bitrise.io/app/4cbfc6308f5c80e5/status.svg?token=FXilfAPD1nT8M1QzlfBNqA&branch=kotlin)
```
Comment (Wadeewee) [inline RxJavaTemplate/README.md:1]: ☝️ Can we remove this? 🙏

**Comment by Tuubz** [comment]
Updated 🚀

**Review by chornerman — APPROVED** [review]

**Review by Wadeewee — APPROVED** [review]

**Review by minhnimble — APPROVED** [review]

File: RxJavaTemplate/README.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Android Templates: Kotlin
+- Our optimized Android templates used in our kotlin android projects
+
+### Setup
+- Clone the project
+- Checkout our main development branch `kotlin`
+- Run the project with Android Studio
+
+
```
[the remark this answers is not in the document]
Comment (Tuubz) [reply RxJavaTemplate/README.md:9]: My bad, will remove 🙏

**Comment by hoangnguyen92dn** [comment]
@Tuubz Please resolve the conflicts 😄

**Comment by Tuubz** [comment]
Updated 🚀

**Comment by Tuubz** [comment]
@minhnimble @Wadeewee @chornerman Please help me recheck, I had to re-do everything. Rebasing didn't really work out  🙇

File: .gitignore [hunk 3]
```diff
@@ -1,57 +1,8 @@
...
-
-# Fastlane
-fastlane/report.xml
+.idea
+.gradle
+/RxJavaTemplate/local.properties
+.DS_Store
+build/
+/captures
+.externalNativeBuild
```
Comment (chornerman) [inline .gitignore:8]: Just curious should we add a blank space here? 🤔

File: gradle-wrapper.properties [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#Tue Jan 12 10:52:22 ICT 2021
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
+org.gradle.parallel=true
```
Comment (chornerman) [inline gradle-wrapper.properties:7]: Same here 🤔

File: RxJavaTemplate/.gitignore [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+# Google Services (e.g. APIs or Firebase)
+#google-services.json
+
+# Freeline
+freeline.py
+freeline/
+freeline_project_description.json
+
+# Fastlane
+fastlane/report.xml
```
Comment (chornerman) [inline RxJavaTemplate/.gitignore:57]: Same here 🤔

**Comment by Tuubz** [comment]
@chornerman Updated 🚀 🚀

**Review by chornerman — APPROVED** [review]

**Review by Wadeewee — APPROVED** [review]

**Review by luongvo — APPROVED** [review]
💯

**Pull request merged**